bin/pycbc_convertinjfiletohdf

```diff
@@ -0,0 +1,181 @@
+from ligo.lw import utils as ligolw_utils, lsctables
+
+
+# Handlers for alternative injection formats
+
+def legacy_approximant_name(apx):
+    """Convert the old style xml approximant name to a name
+    and phase_order.
+    """
+    apx = str(apx)
+    try:
+        order = sim.GetOrderFromString(apx)
+    except:
+        print("Warning: Could not read phase order from string, using default")
+        order = -1
+    name = sim.GetStringFromApproximant(sim.GetApproximantFromString(apx))
+    return name, order
+
+
+class XMLInjectionSet(object):
+
+    """Reads injections from LIGOLW XML files
+
+    Parameters
+    ----------
+    sim_file : string
+        Path to a LIGOLW XML file containing a SimInspiralTable
+        with injection definitions.
+
+    Attributes
+    ----------
+    indoc
+    table
+    """
+
+    def __init__(self, sim_file):  
+        self.indoc = ligolw_utils.load_filename(
+            sim_file, False, contenthandler=LIGOLWContentHandler)
+        self.table = lsctables.SimInspiralTable.get_table(self.indoc)
+
+    def end_times(self):
+        """Return the end times of all injections"""
+        return [inj.time_geocent for inj in self.table]
+
+    def pack_data_into_pycbc_format_input(self):
+        data = {}
+        for key in xinj.table[0].__slots__:
+            # Some XML files can have empty columns which are read as None in
+            # python. For these cases we ignore the columns so they do not
+            # appear in the output HDF file. (Such XML files cannot currently
+            # be read by LALSuite C code.) 
+            if getattr(xinj.table[0], key) is not None:
+                data[str(key)] = numpy.array(
+                    [getattr(t, key) for t in xinj.table]
+                )
+
+        for k in ['simulation_id', 'process_id']:
+            a = data.pop(k)
+
+        data['approximant'], data['phase_order'] = numpy.array(
+                [legacy_approximant_name(wf) for wf in data['waveform']]
+        ).T
+        data['tc'] = (
+            data['geocent_end_time'] + 1e-9 * data['geocent_end_time_ns']
+        )
+        data['dec'] = data['latitude']
+        data['ra'] = data['longitude']
+        return data
+
+
+class LVKNewStyleInjectionSet(object):
+    """Reads injections from new-style LVK injection files
+    """
+
+    translation_dict = {
+        'mass1_det': 'mass1',
+        'mass2_det': 'mass2',
+        'f22_ref_spin': 'fref', # ????
+        't_co_gps': 'tc',
+        'd_lum': 'distance',
+        'f22_start': 'f_lower',
+        'cbc_model': 'approximant',
+        'longAscNodes': 'FIXME', # FIXME
+        'eccentricity': 'FIXME', # FIXME (check eccentricity params!)
+        'meanPerAno': 'FIXME', # FIXME
+        't_co_gps_add': 'FIXME', # FIXME (don't understand why this is needed?)
+        'ModeArray': 'FIXME', # FIXME (Will need special handling. Need an example)
+        'ModeArrayJframe': 'FIXME', # FIXME (Does this do anything useful?)
```

Comment:
Same explanation here. This may or may not be 'useful' but it corresponds to a LALSimInspiral parameters slot, so someone at some point thought it might be.